### PR TITLE
Inline js runtime bundle

### PR DESCRIPTION
**Pull request by yishengjiang99**
Updated PR to make it work for cdn/mock-cloud mode as well. See attached screenshots:
pr-screen1
pr-screen2

### CODE REVIEW

**Comment by jchip** [comment]
in the page source, line `31`, it's marking only 2 bundles `1, 4` as loaded.  it should've also mark the bundle for runtime as loaded.

**Comment by yishengjiang99** [comment]
Ok.

**Comment by yishengjiang99** [comment]
> in the page source, line `31`, it's marking only 2 bundles `1, 4` as loaded. it should've also mark the bundle for runtime as loaded.
see updated PR and screenshot for call to markBundlesLoaded.

pr-screenshot3-xarcV1-markloaded

File: packages/subapp-web/lib/init.js [hunk 1]
```diff
@@ -34,12 +30,32 @@ module.exports = function setup(setupContext) {
...
+  let inlineRuntimeJS = "";
+  let rtJsCdnEntryPoint;
+  if (process.env.NODE_ENV === "production") {
+    rtJsCdnEntryPoint = Object.keys(cdnJsBundles).filter(ep =>
+      cdnJsBundles[ep].includes("runtime.bundle")
+    )[0];
+    const runtimeFullPath =
+      rtJsCdnEntryPoint &&
+      cdnJsBundles[rtJsCdnEntryPoint] &&
+      Path.join(Path.resolve("dist"), "js", Path.basename(cdnJsBundles[rtJsCdnEntryPoint]));
```
Comment (jchip) [inline packages/subapp-web/lib/init.js:42]: join and resolve together is redundant: https://nodejs.org/api/path.html#path_path_resolve_paths

File: packages/subapp-web/lib/init.js [hunk 2]
```diff
@@ -34,12 +30,32 @@ module.exports = function setup(setupContext) {
...
+
   const webSubAppJs = `<script id="bundleAssets" type="application/json">
 ${JSON.stringify(bundleAssets)}
 </script>
 <script>/*LJ*/${loadJs}/*LJ*/
+/*rt*/${inlineRuntimeJS} /*rt*/
 ${clientJs}
 ${cdnJs}
+${
+  inlineRuntimeJS && rtJsCdnEntryPoint && `window.xarcV1.markBundlesLoaded([${rtJsCdnEntryPoint}]);`
```
Comment (jchip) [inline packages/subapp-web/lib/init.js:56]: please make this code and line 52 more concise.

File: packages/subapp-web/lib/init.js [hunk 3]
```diff
@@ -34,12 +30,32 @@ module.exports = function setup(setupContext) {
     basePath: ""
   };
 
+  let inlineRuntimeJS = "";
+  let rtJsCdnEntryPoint;
+  if (process.env.NODE_ENV === "production") {
+    rtJsCdnEntryPoint = Object.keys(cdnJsBundles).filter(ep =>
+      cdnJsBundles[ep].includes("runtime.bundle")
```
Comment (jchip) [inline packages/subapp-web/lib/init.js:37]: use `startsWith`

File: packages/subapp-web/lib/init.js (hunk 2 above)
Comment (yishengjiang99) [reply packages/subapp-web/lib/init.js]: You are right. see updated PR.
image
image

File: packages/subapp-web/lib/init.js (hunk 3 above)
Comment (yishengjiang99) [reply packages/subapp-web/lib/init.js:36]: can't... some clap build presets results in 'js/runtime.bundle.js' in this particular variable.

File: packages/subapp-web/lib/init.js [hunk 4]
```diff
@@ -34,14 +29,28 @@ module.exports = function setup(setupContext) {
...
   const webSubAppJs = `<script id="bundleAssets" type="application/json">
 ${JSON.stringify(bundleAssets)}
 </script>
 <script>/*LJ*/${loadJs}/*LJ*/
+/*rt*/${inlineRuntimeJS} /*rt*/
 ${clientJs}
 ${cdnJs}
-</script>
-`;
+window.xarcV1.markBundlesLoaded([${runtimeEntryPoints.join(",")}]);
```
Comment (jchip) [inline packages/subapp-web/lib/init.js:52]: what if `runtimeEntryPoints` has nothing?

File: packages/subapp-web/lib/init.js (hunk 3 above)
Comment (jchip) [reply packages/subapp-web/lib/init.js:36]: then you are not referencing the right things

File: packages/subapp-web/lib/init.js (hunk 4 above)
Comment (yishengjiang99) [reply packages/subapp-web/lib/init.js:52]: it would call ```window.xarcV1.markBundlesLoaded([]);``` which doesn't seem to throw an exception.

Comment (yishengjiang99) [reply packages/subapp-web/lib/init.js]: I changed to using ```window.xarcV1.markBundlesLoaded(${JSON.stringify(runtimeEntryPoints)});
```

File: packages/subapp-web/lib/init.js (hunk 3 above)
Comment (yishengjiang99) [reply packages/subapp-web/lib/init.js]: ok i changed to using `    runtimeEntryPoints = Object.keys(assets.chunksById.js).filter(ep =>
      assets.chunksById.js[ep].startsWith("runtime.bundle")
    );`

**Comment by yishengjiang99** [comment]
pr-june-1-dev-ss
pr-june-1-prod-ss
image

@jchip updated pr per conversation half hour ago. see screenshot

**Pull request merged**